cadence/contracts/services/AxelarGovernanceService.cdc

```diff
@@ -0,0 +1,107 @@
+import IAxelarExecutable from "../AxelarGateway.cdc"
+import AxelarGateway from "../AxelarGateway.cdc"
+
+pub contract AxelarGovernanceService: IAxelarExecutable{
+    access(all) let gateway: Address
+    access(all) let governanceChain: String
+    access(all) let governanceAddress: String
+    access(all) let minimumTimeDelay: UInt64
+
+    access(all) resource Proposal{
+        access(self) let id: String
+        access(self) let proposedCode: String
+        access(self) let target: Address
+        access(self) let timeCreated: UInt64
+        access(self) let timeToExecute: UInt64
+        access(self) let executed: Bool
+
+        init(id: String, proposedCode: String, target: Address, timeToExecute: UInt64){
+            self.id = id
+            self.proposedCode = proposedCode
+            self.target = target
+            self.timeCreated = UInt64(getCurrentBlock().timestamp)
+            self.executed = false
+
+            if timeToExecute < UInt64(getCurrentBlock().timestamp) + self.minimumTimeDelay {
+                self.timeToExecute = UInt64(getCurrentBlock().timestamp) + self.minimumTimeDelay
+            } else {
+                self.timeToExecute = timeToExecute
+            }
+        }
+
+    }
+
+    /**********\
+    |* Events *|
+    \**********/
+
+    pub event ProposalScheduled(
+        proposalHash: String,
+        target: Address,
+        proposedCode: String,
+        eta: UInt64
+    )
+
+    pub event ProposalCancelled(
+        proposalHash: String,
+        target: Address,
+        proposedCode: String,
+        eta: UInt64
+    )
+
+    pub event ProposalExecuted(
+        proposalHash: String,
+        target: Address,
+        proposedCode: String,
+        timestamp: UInt64
+    )
+
+
+    init(gateway: Address, governanceChain: String, governanceAddress: String, minimumTimeDelay: UInt64){
+        self.gateway = gateway
+        self.governanceChain = governanceChain
+        self.governanceAddress = governanceAddress
+        self.minimumTimeDelay = minimumTimeDelay
+    }
+
+    access(self) let proposals: {String: @Proposal}
```

Comment:
Q: What sort of scale do we expect these proposals to reach? Asking because mappings can introduce scaling considerations. Also wondering if there should be getter and/or removal methods

np: move up with other contract fields

Reply:
I believe the scale will be minimal, from what i remember from axelar, it sounded like they would submit a proposal once a week at most.  The proposals should also be destroyed when executed.